src/univariate/discrete/geometric.jl

```diff
@@ -124,7 +124,7 @@ end
 
 function mgf(d::Geometric, t::Real)
     p = succprob(d)
-    p / (expm1(-t) + p)
+    p / (expm1(-t) + p) - t
```

Comment:
Shouldn't this be reverted to
```suggestion
    p / (expm1(-t) + p)
```
(see e.g. https://en.wikipedia.org/wiki/Moment-generating_function#Examples)?

Reply:
I guess the tests for `Geometric` pass even with the (IMO) incorrect version since `-t` does not affect the derivative at 0.

Reply:
`-t` adds a `-1` to the derivative. This corresponds to shifting the expectation value by `1`. Which corresponds to chaning the `Geometric` parametrization from `k-1` to `k`. See also my comments below.

Reply:
Ah, I completely missed the different parameterization here. However, AFAIK the linear transformation Y := X - 1 of a random variable X should result in M_Y(t) = exp(-t) * M_X(t) instead of a shift by -t. So it seems it should be something like in https://github.com/JuliaStats/Distributions.jl/pull/1604/files/1b75633699749011a03a6b0ec85c335a8281073a#r945207774 (without the `^r`), e.g.,
```suggestion
    p / (1 - (1 - p) * exp(t))
```
That would also correspond to the formula given on Wikipedia (https://en.wikipedia.org/wiki/Geometric_distribution, right column).

Reply:
Yes you are right, I was thinking about `cgf = log mgf` where `exp(t)` becomes `t`.